### PR TITLE
Expand shim for processor and outputs.

### PR DESCRIPTION
**Pull request by ssoroka**
Expands execd shim support for processors and outputs. 
Execd lets you write custom input, processor, and output plugins in any language. The execd shim lets you run plugins written in Go without changing the structure, regardless of whether they're run as internal plugins compiled into Telegraf, or external plugins compiled as a separate binary and run through execd.

### CODE REVIEW

File: plugins/common/shim/config.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	if conf.Input != nil {
+		if err = s.AddInput(conf.Input); err != nil {
+			return fmt.Errorf("Failed to add Input: %w", err)
+		}
+	} else if conf.Processor != nil {
+		if err = s.AddStreamingProcessor(conf.Processor); err != nil {
+			return fmt.Errorf("Failed to add Processor: %w", err)
+		}
+	} else if conf.Output != nil {
+		if err = s.AddOutput(conf.Output); err != nil {
```
Comment (reimda) [inline plugins/common/shim/config.go:43]: Does this let people have multiple inputs, processors, and outputs in the same shim?  I guess I expected one plugin of any type per shim process.

Comment (ssoroka) [reply plugins/common/shim/config.go:43]: yeah, you can only have one.

**Comment by ssoroka** [comment]
rebase master to resolve conflicts

**Comment by programmer04** [comment]
Merging this PR will fix some data races reported in #7729 too.

File: plugins/common/shim/output.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		}
+	}
+
+	s.Output = output
+	return nil
+}
+
+func (s *Shim) RunOutput() error {
+	// TODO: ? Need to support multiple parsers, but not clear how.
+	parser, err := parsers.NewInfluxParser()
```
Comment (danielnelson) [inline plugins/common/shim/output.go:26]: I would say no need, the shim abstracts the parsing aspect so you only need to deal with telegraf.Metric as a user.

File: plugins/common/shim/output.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	}
+	defer s.Output.Close()
+
+	var m telegraf.Metric
+
+	scanner := bufio.NewScanner(s.stdin)
+	for scanner.Scan() {
+		m, err = parser.ParseLine(scanner.Text())
+		if err != nil {
+			fmt.Fprintf(s.stderr, "Failed to parse metric: %s\b", err)
```
Comment (danielnelson) [inline plugins/common/shim/output.go:43]: return/continue?

File: plugins/common/shim/output.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		}
+		if err = s.Output.Write([]telegraf.Metric{m}); err != nil {
+			fmt.Fprintf(s.stderr, "Failed to write metric: %s\b", err)
```
Comment (danielnelson) [inline plugins/common/shim/output.go:46]: `\b`?

File: plugins/common/shim/output.go (hunk 2 above)
Comment (ssoroka) [reply plugins/common/shim/output.go:26]: Would be nice for non-shim users who are more familiar with JSON, but that's a deeper topic for another time

**Pull request closed without merging**